src/h5web/providers/h5core/models.ts

```diff
@@ -0,0 +1,45 @@
+import type { EntityKind } from '../models';
+
+export type H5CoreAttrResponse = Record<string, unknown>;
+
+export type H5CoreDataResponse = unknown;
+
+export interface H5CoreMetaResponse {
```

Comment:
If I remember correctly our discussions, it is easier for TS to have a base interface like `H5CoreMetaResponse` like this with extensions rather than an union type `H5CoreMetaResponse = H5CoreDatasetMetaReponse | H5CoreGroupMetaReponse | ...`. 

Could you confirm that this is right ?

Reply:
It's better for type-checking performance, but it requires the use of typeguards to narrow down the types, which is fine here. With type unions, if all types are mutually exclusive in some way (e.g. an enum property), then TypeScript can narrow down the type with a simple `if` or `switch`.